OpenEphys.Onix/OpenEphys.Onix/NeuropixelsV2RegisterContext.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections;
+
+namespace OpenEphys.Onix
+{
+    class NeuropixelsV2RegisterContext : I2CRegisterContext
+    {
+        public NeuropixelsV2RegisterContext(I2CRegisterContext other, uint i2cAddress)
+            : base(other, i2cAddress)
+        {
+        }
+
+        public NeuropixelsV2RegisterContext(DeviceContext deviceContext, uint i2cAddress)
+            : base(deviceContext, i2cAddress)
+        {
+        }
+
+        public void WriteConfiguration(NeuropixelsV2QuadShankProbe probe)
+        {
+            var baseBits = GenerateBaseBits(probe);
+            WriteShiftRegister(NeuropixelsV2Definitions.SR_CHAIN5, baseBits[0]);
+            WriteShiftRegister(NeuropixelsV2Definitions.SR_CHAIN6, baseBits[1]);
+
+            var shankBits = GenerateShankBits(probe);
+            WriteShiftRegister(NeuropixelsV2Definitions.SR_CHAIN1, shankBits[0]);
+            WriteShiftRegister(NeuropixelsV2Definitions.SR_CHAIN2, shankBits[1]);
+            WriteShiftRegister(NeuropixelsV2Definitions.SR_CHAIN3, shankBits[2]);
+            WriteShiftRegister(NeuropixelsV2Definitions.SR_CHAIN4, shankBits[3]);
+
+        }
+
+        // Bits go into the shift registers MSB first
+        // This creates a *bit-reversed* byte array from a bit array
+        static byte[] BitArrayToBytes(BitArray bits)
+        {
+            if (bits.Length == 0)
+            {
+                throw new ArgumentException("Shift register data is empty", nameof(bits));
+            }
+
+            var bytes = new byte[(bits.Length - 1) / 8 + 1];
+            bits.CopyTo(bytes, 0);
+
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                // NB: http://graphics.stanford.edu/~seander/bithacks.html
+                bytes[i] = (byte)((bytes[i] * 0x0202020202ul & 0x010884422010ul) % 1023);
+            }
+
+            return bytes;
+        }
+
+        // TODO: NeuropixelsV2Definitions.STATUS always fails.
```

Comment:
```suggestion
        // TODO: NeuropixelsV2.STATUS always fails.
```

If the above recommendation to rename the internal static class is accepted, this should be renamed too.